src/Employer/Employer.Web/Filters/EmployerAccessCheckFilter.cs

```diff
@@ -0,0 +1,264 @@
+            // or redirect to the Levy Declaration page for levy employers
+            string employerAccountType = await GetEmployerAccountType(context, userId, employerAccountId);
+            if (string.Compare(employerAccountType, "levy", true) == 0)
+            {
+                context.Result = new RedirectToRouteResult(RouteNames.LevyDeclaration_Get, new { employerAccountId });
+            }
+            else
+            {
+                throw new BlockedEmployerException($"Employer account '{employerAccountId}' is blocked");
+            }
+        }
+
+        private async Task<string> GetEmployerAccountType(ActionExecutingContext context, string userId, string employerAccountId)
+        {
+            string result;
+            if (!GetCookieValueForUserAndEmployer(context.HttpContext, 
+                userId: userId,
+                employerAccountId: employerAccountId,
+                readValue: _employerAccountTypeCookieWriter.GetCookieFromRequest,
+                result: out result))
+            {
+                AccountDetailViewModel accountDetails = await _recruitVacancyClient.GetEmployerAccountDetailsAsync(employerAccountId);
+                result = accountDetails.ApprenticeshipEmployerType;
+
+                if (!_allowedAccountTypes.Any(x => string.Compare(accountDetails.ApprenticeshipEmployerType, x, true) == 0))
+                    throw new BlockedEmployerException($"Unknown account type {accountDetails.ApprenticeshipEmployerType}");
+
+                _employerAccountTypeCookieWriter.WriteCookie(context.HttpContext.Response, userId, employerAccountId, result);
+            }
+            return result;
+        }
+
+        private async Task<bool> WasHandledByLevyDeclaration(
+            ActionExecutingContext context,
+            ActionExecutionDelegate next,
+            string userId,
+            string employerAccountId,
+            bool requestIsForALevyPage)
+        {
+            bool hasLevyDeclaration = false;
+            if (GetCookieValueForUserAndEmployer(
+                context.HttpContext,
+                userId: userId,
+                employerAccountId: employerAccountId,
+                readValue: _levyCookieWriter.GetCookieFromRequest,
+                result: out string hasLevyDeclarationAsString))
+            {
+                hasLevyDeclaration = bool.Parse(hasLevyDeclarationAsString);
+            }
+            else
+            {
+                hasLevyDeclaration = await HasStoredLevyDeclaration(
+                    employerAccountId: employerAccountId,
+                    userId: userId);
+                _levyCookieWriter.WriteCookie(context.HttpContext.Response, userId, employerAccountId, hasLevyDeclaration);
+            }
+
+            // No Levy declaration = not handled
+            if (!hasLevyDeclaration)
+                return false;
+
+            // Don't allow Levy pages if Levy is already declared
```

Comment:
If they have Levy declared then we should allow access to Levy pages right

Reply:
Not according to the original code, no